plugins/generator-1.18.1/forge-1.18.1/templates/elementinits/features.java.ftl

```diff
@@ -46,24 +46,30 @@ package ${package}.init;
 			REGISTRY.put(${feature.getModElement().getName()}Feature.FEATURE, new FeatureRegistration(GenerationStep.Decoration.UNDERGROUND_ORES,
 						${feature.getModElement().getName()}Feature.GENERATE_BIOMES, ${feature.getModElement().getName()}Feature.CONFIGURED_FEATURE,
 						${feature.getModElement().getName()}Feature.PLACED_FEATURE));
+		<#elseif feature.getModElement().getTypeString() == "structure">
+			REGISTRY.put(${feature.getModElement().getName()}Feature.FEATURE, new FeatureRegistration(GenerationStep.Decoration.
+						<#if feature.spawnLocation == "Air">RAW_GENERATION<#elseif feature.spawnLocation == "Underground">UNDERGROUND_STRUCTURES<#else>SURFACE_STRUCTURES</#if>,
+						${feature.getModElement().getName()}Feature.GENERATE_BIOMES, ${feature.getModElement().getName()}Feature.CONFIGURED_FEATURE,
+						${feature.getModElement().getName()}Feature.CONFIGURED_FEATURE.placed(List.of())));

```

Comment:
You should keep placed feature in the feature class, not here, to keep all elements consistent